app/scripts/controllers/pod.js

```diff
@@ -87,6 +87,58 @@ angular.module('openshiftConsole')
       }
     };
 
+    /**
+     *  Will set the newTerm's isActive/isUsed to true, while deactivating the old's
+     */
+    $scope.onTerminalSelectChange = function(newTerm) {
+      // Deactivate all terminals
+      _.each($scope.containerTerminals, function(term) {
+        term.isActive = false;
+      });
+
+      newTerm.value.isActive = true;
+      newTerm.value.isUsed = true;
+    };
+
+    $scope.tabSelect = function() {
+      $scope.terminalTabWasSelected = true;
+    };
+
+    var makeTerminals = function(containerStatuses, existingTerms) {
+      var firstRun = false;
```

Comment:
although you'll still need to initialize existingTerms.  thats another one-liner
```
var existingTerms = existingTerms || {};
```
because javascript unfortunately doesnt have an `||=` operator